File: VB/MvcPivotChartIntegration/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace MvcApplication1.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function


		<ValidateInput(False)>
		Public Function PivotGridPartial() As ActionResult
			Return PartialView("_PivotGridPartial", GetData())
		End Function

		Public Function ChartPartial() As ActionResult
			Return PartialView("_ChartPartial", GetData())
		End Function
		Public Shared Function GetData() As DataTable
			Dim names() As String = { "Aaa", "Bbb", "Ccc", "Ddd" }

			Dim dt As New DataTable()
			Dim _r As New Random()
			dt.Columns.Add("Date", GetType(Date))
			dt.Columns.Add("Name", GetType(String))
			dt.Columns.Add("Value", GetType(Decimal))
			For i As Integer = 0 To 27
				dt.Rows.Add(Date.Today.AddHours(6 * i), names(i Mod 4), i Mod 5 * 6 + i Mod 3 * 7 + 20)
			Next i
			Return dt
		End Function
	End Class
End Namespace

File: CS/MvcPivotChartIntegration/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }


        [ValidateInput(false)]
        public ActionResult PivotGridPartial()
        {
            return PartialView("_PivotGridPartial", GetData());
        }

        public ActionResult ChartPartial()
        {
            return PartialView("_ChartPartial", GetData());
        }
        public static DataTable GetData()
        {
            string[] names = new string[] { "Aaa", "Bbb", "Ccc", "Ddd" };
            
            DataTable dt = new DataTable();
            Random _r = new Random();
            dt.Columns.Add("Date", typeof(DateTime));
            dt.Columns.Add("Name", typeof(string));
            dt.Columns.Add("Value", typeof(decimal));
            for (int i = 0; i < 28; i++)
                dt.Rows.Add(DateTime.Today.AddHours(6 * i), names[i % 4], i % 5 * 6 + i % 3 * 7 + 20);
            return dt;
        }
    }
}
